schematic/store/synapse.py

```diff
@@ -777,18 +788,47 @@ def fill_in_entity_id_filename(
         # the columns Filename and entityId are assumed to be present in manifest schema
         # TODO: use idiomatic panda syntax
         if dataset_files:
```

Comment:
You could flip this `if` check and lower the congnative complexity of this function. Like:

```
  if not dataset_files:
      manifest = manifest.fillna("")
      return dataset_files, manifest

# The rest of the function
.
.
.
.
manifest = manifest.fillna("")
return dataset_files, manifest
    
```

The idea here is that you are failing fast with the negative conditions ie: If you are providing things that I am not expecting I am going to return as soon as possible. That way you don't have significantly nested code; In the rest of the function you can guarantee that `dataset_files` has a truthy value in it.

Reply:
That makes a lot of sense @BryanFauble thank you